numba/cpython/mathimpl.py

```diff
@@ -455,3 +458,26 @@ def gcd(a, b):
 
 
 lower(math.gcd, types.Integer, types.Integer)(gcd_impl)
+
+
+# -----------------------------------------------------------------------------
+# Equality tests implemented on Floats
+# Implemented based on this feature request
+# - https://github.com/numba/numba/issues/5977
+# This does not implement the * in
+# https://docs.python.org/3/library/math.html#math.isclose,
+# which limits the number of provided positional arguments to 2.
+@overload(math.isclose)
+def math_isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
+    if not isinstance(a, types.Number) or not isinstance(b, types.Number):
+        raise TypingError("All arguments must be scalars.")
+    if not isinstance(rel_tol, types.Number) or not isinstance(abs_tol, types.Number):
+        raise TypingError("All arguments must be scalars.")
+
+    def impl(a, b, rel_tol=1e-09, abs_tol=0.0):
+        if a == b:
+            return True
+        abs_diff = np.abs(a - b)
+        min_tolerance = np.maximum(rel_tol * np.maximum(a, b), abs_tol)
+        return abs_diff <= min_tolerance
```

Comment:
Perhaps take a look at the CPython implementation for how to do this, Numba needs to match this and it'd be a good place to gain insight in to what to additionally consider. https://github.com/python/cpython/blob/8e19c8be87017f6bef8e4c936b1e6ddacb558ad2/Modules/mathmodule.c#L2931-L2971